cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -196,4 +196,34 @@ SensorBridge* MapBuilderBridge::sensor_bridge(const int trajectory_id) {
   return sensor_bridges_.at(trajectory_id).get();
 }
 
+visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodesList() {
+  visualization_msgs::MarkerArray trajectory_nodes_list;
+
+  for (int trajectory_id = 0;
+       trajectory_id < map_builder_.num_trajectory_builders();
+       ++trajectory_id) {
+    const auto trajectory_nodes =
+        map_builder_.sparse_pose_graph()->GetTrajectoryNodes();
+
+    int marker_id = 0;
+    for (const auto& single_trajectory : trajectory_nodes) {
+      for (const auto& node : single_trajectory) {
+        visualization_msgs::Marker marker;
+        marker.id = marker_id++;
+        marker.type = visualization_msgs::Marker::ARROW;
```

Comment:
Reading <http://wiki.ros.org/rviz/DisplayTypes/Marker>, I wonder how efficient rendering each trajectory node as an `ARROW` marker is. If we're okay with losing the orientation information (the arrows already give us only 2 of the angles, not all 3), we could use a much more efficient `LINE_STRIP`, i.e. render each complete trajectory as one marker.